pygpsclient/scatter_frame.py

```diff
@@ -0,0 +1,217 @@
+
+This generates a scatterplot of positions, centered on the average
+position.
+
+Created 23 March 2023
+
+:author: Nathan Michaels
+:copyright: Qualinx B.V.
+:license: BSD 3-Clause
+"""
+from collections import namedtuple
+from statistics import mean
+import math
+from tkinter import Frame, font, BOTH, YES
+from geographiclib.geodesic import Geodesic
+from pygpsclient.globals import WIDGETU2, BGCOL, FGCOL
+from pygpsclient.skyview_frame import Canvas
+
+
+Point = namedtuple("Point", ["lat", "lon"])
+geo = Geodesic.WGS84
+
+
+# pylint: disable=too-many-instance-attributes
+class ScatterViewFrame(Frame):
+    """
+    Scatterplot view frame class.
+    """
+
+    def __init__(self, app, *args, **kwargs):
+        """
+        Constructor.
+
+        :param Frame app: reference to main tkinter application
+        :param args: Optional args to pass to Frame parent class
+        :param kwargs: Optional kwargs to pass to Frame parent class
+        """
+        self.__app = app
+        self.__master = self.__app.appmaster
+
+        Frame.__init__(self, self.__master, *args, **kwargs)
+
+        def_w, def_h = WIDGETU2
+
+        self.dot_col = "#8888FF"
+        self.width = kwargs.get("width", def_w)
+        self.height = kwargs.get("height", def_h)
+        self.fg_col = FGCOL
+        self.points = []
+        self.one_meter = 1
+        self.mean = None
+        self._body()
+        self.bind("<Configure>", self._on_resize)
+
+    def _body(self):
+        """Set up frame and widgets."""
+
+        self.grid_columnconfigure(0, weight=1)
+        self.grid_rowconfigure(0, weight=1)
+        self.canvas = Canvas(self, width=self.width, height=self.height, bg=BGCOL)
+        self.canvas.pack(fill=BOTH, expand=YES)
+
+    def _on_resize(self, event):  # pylint: disable=unused-argument
+        """
+        Resize frame
+
+        :param event: resize event
+        """
+
+        self.width, self.height = self.get_size()
+        self.init_graph()
+        self.redraw()
+
+    def get_size(self):
+        """
+        Get current canvas size.
+
+        :return window size (width, height)
+        """
+
+        self.update_idletasks()  # Make sure we know about resizing
+        width = self.canvas.winfo_width()
+        height = self.canvas.winfo_height()
+        return (width, height)
+
+    def _draw_mean(self, lbl_font):
+        """
+        Draw the mean position in the corner of the plot. Uses
+        self.mean as the position to draw.
+
+        :param lbl_font: Font to use.
+        """
+
+        if self.mean is None:
+            return
+
+        height = lbl_font.metrics("linespace")
+        lat = f"Lon {abs(self.mean.lat):<012.10}"
+        lon = f"Lat {abs(self.mean.lon):<012.10}"
+        width = max(lbl_font.measure(lat), lbl_font.measure(lon))
+        self.canvas.create_text(
+            width, height, text=lat, fill=self.fg_col, font=lbl_font
+        )
+        self.canvas.create_text(
```

Comment:
by using the absolute value you're losing the decimal sign and hence the N/S differentiation.